src/test/run-pass/generator/iterator-count.rs

```diff
@@ -18,7 +18,7 @@ impl<T: Generator<Return = ()>> Iterator for W<T> {
     type Item = T::Yield;
 
     fn next(&mut self) -> Option<Self::Item> {
-        match self.0.resume() {
+        match unsafe { self.0.resume() } {
```

Comment:
This impl isn't actually sound anymore since the iterator can move-- can you leave a comment?